gomanifest/internal/deps.go

```diff
@@ -0,0 +1,100 @@
+package internal
+
+import (
+	"encoding/json"
+	"io"
+	"os/exec"
+	"strings"
+
+	"github.com/rs/zerolog/log"
+)
+
+/* Structure required for parsing `go list -deps -json ./...` json parsing */
+
+// DepModule ... Module structure from deps json
+type DepModule struct {
+	Path    string     `json:"Path"`
+	Main    bool       `json:"Main"`
+	Version string     `json:"Version"`
+	Replace *DepModule `json:"Replace"`
+}
+
+// DepPackage ... Package structure from deps json
+type DepPackage struct {
+	Root       string    `json:"Root"`
+	ImportPath string    `json:"ImportPath"`
+	Module     DepModule `json:"Module"`
+	Standard   bool      `json:"Standard"`
+	Imports    []string  `json:"Imports"`
+	Deps       []string  `json:"Deps"`
+}
+
+// GoListCmd ... Go list command structure.
+type GoListCmd struct {
+	CWD string
+}
+
+// Run ... Actual function that executes go list command and returns output as string.
+func (goListCmd GoListCmd) Run() (string, error) {
+	GoListGoListDeps := exec.Command("go", "list", "-json", "-deps", "./...")
+	GoListGoListDeps.Dir = goListCmd.CWD
+	output, err := GoListGoListDeps.Output()
+
+	if err != nil {
+		return "", err
+	}
+	return string(output), nil
+}
```

Comment:
Since now you are doing a stream json processing, using [cmd.StdoutPipe](https://golang.org/pkg/os/exec/#Cmd.StdoutPipe) makes more sense than cmd.Output. StdoutPipe will give you io.Reader which you can directly pass to parseDepsJSON.
```suggestion
func (goListCmd GoListCmd) Run() (io.ReadCloser, error) {
       goList := exec.Command("go", "list", "-json", "-deps", "./...")
       goList.Dir = goListCmd.CWD
       defer goList.Start()
       return goList.StdoutPipe()
 }
```

Reply:
This is done.